### DISCUSSION TITLE
Cloud Segmentation

### DISCUSSION THREAD

**Discussion by Kcok** [ultralytics/hub#810] [Q&A]
Hi everybody,

I've trained a segmentation model using YOLOv9-seg as starting point for Cloud Detection and segmentation. I've trained with and without data augmentation. 500 Epochs with a patience of 15 epochs. The data set I have used is the one used for this challenge: "[](https://drivendata.co/blog/cloud-cover-winners/)" with 63255 good labelled data. which I've split 70 - 15 - 15.
The data was normalised, and the labels converted from binary mask to YOLO text format.

Despite all of this, the generated model can barely detect clouds with a probability higher than 20%, which makes the model actually not reliable. The model have specially problems with "big" clouds (covering 90% or more from the image). After a deeper data analysis, the percentage of images with high cloud coverage (cloud > 90%) is well represented in the training data set with about 46% of the images.

**Is there something I am missing? or something I can do to get better accuracy while doing detection?** I have to say, that when the detection is there, the segmentation works like charm.

Thanks in advance.

**Comment by Kcok** [comment]
[https://drivendata.co/blog/cloud-cover-winners/](url)

**Reply by mirage-4000** [reply]
try adding you own image in the dataset, this problem may occur because your camera is to different from the camera wich take the image ( maybe the camera work better with light fluctuation, maybe it have a better exposure etc ).try with 100 epochs, the model may have done overfitting , also try changing the threshold to see what's the best value . if the problem persists ask a question in the support tab on the website, if you can give more information it's may help them to fixe issue ( like map )
